cql3/statements/index_target.cc

```diff
@@ -71,6 +73,22 @@ index_target::target_type index_target::from_sstring(const sstring& s)
     throw std::runtime_error(format("Unknown target type: {}", s));
 }
 
+index_target::target_type index_target::from_target_string(const sstring& target) {
+    std::cmatch match;
+    if (std::regex_match(target.data(), match, target_regex)) {
+        return index_target::from_sstring(match[1].str());
+    }
+    return target_type::regular_values;
+}
+
+sstring index_target::column_name_from_target_string(const sstring& target) {
+    std::cmatch match;
+    if (std::regex_match(target.data(), match, target_regex)) {
+        return match[2].str();
```

Comment:
It would, but I don't immediately see how to do it nicely. There could be one function that returns a pair, but `index::index` uses those values to initialize members - I would have to remove those members from member initializer list and instead initialize them in constructor body, which seems like less elegant solution. Would you like me to do that, or maybe you have some other idea?

Reply:
I also think that performance overhead shouldn't matter too much, `index::index` is created once per secondary index per shard.